### PR TITLE
Javascript fixes for dropdown menus hover + escape key functionality

### PR DESCRIPTION
**Pull request by judaw**
Recs from Alison:
- [ ] Put this branch on a cddemo multidev, so it can be reviewed IRL (and add a link to that multidev here).
- [ ] Review changes given what we've learned since the HR site launched, about drop-down menu behavior differences in NVDA vs. Voiceover (some details in comment thread below; WA team and/or Tone can explain it better).

Question from Alison:
* Do these changes belong in cwd_framework, **_then_** cwd_base?  (sincere question!)
  * If yes, do that, then start new PR over here.
  * P.S. Even if yes, whoever's doing the work can still create a multidev on cddemo and manually put the changes up there to help with testing and review!

### CODE REVIEW

**Comment by alisonjo315** [comment]
Maybe someone could put this branch (or a combination of this branch and #61) on a `cddemo` multidev, so it can be reviewed IRL?

(Cross-posting on #61)

**Comment by alisonjo315** [comment]
TL;DR: Is there a chance the HR drop-down menus were built / evaluated only with Voiceover, not NVDA...?
* CC @sg667 @judaw

----
I apologize if I'm off-base here or out of scope or whatever, I'm mediocre-at-best at using NVDA and my struggle could absolutely be me not the HR site -- but anyway:

While reviewing Engineering, I was trying to figure out how a main menu with drop-downs _**should**_ behave, and I remembered there was a lot of accessibility work done on the HR menu, so I tried out the HR main menu, navigating with a screen reader and keyboard -- specifically with NVDA, because I'm on Windows.

* When I simply listen to the page in order, it only reads top items (good), but I can't actually navigate into the sub-menus -- "down arrow" in NVDA takes me to the next item in the page order (which is the next top-level item in the main menu).  If I press "space" or "enter," it effectively clicks the currently-focused menu link.
* While trying to figure out what I should be doing differently, I also navigated with keyboard -- when I tab to a main menu item, the drop-down does (visually) expand, and an audio prompt says "Jobs: To enter this sub menu, press Down Arrow," but again, when I press down arrow, it goes to the next item in the page order (i.e. the next top-level item in the main menu), it does not take me to the drop-down children.
* (Again, this is normal with NVDA, down/up arrow take you to the next thing on the page, so to speak.)

FWIW, when I'm not using NVDA, when I'm just navigating the HR site via keyboard, I can indeed use "down arrow" to enter the drop-down sub-menus of the main menu.

----

One other thing: The "current page" is visually highlighted, but there's no "aria-current" attribute.  I know this is a separate issue, but I didn't want to bother creating a new issue until someone says "yeah that's a problem."

**Comment by alisonjo315** [comment]
Did the NVDA fixes make it into cwd_framework, or not yet? -- regardless, I wonder if all these fixes belong in cwd_framework, and then the fancy script from cwd_framework is how we should get it into cwd_base?

(Probably a question for the DUX team.)

**Comment by ama39** [comment]
This is a duplicate of the same request in the cwd_framework repo. Fixes for these issues were covered as part of the major main nav work in the 3.0.1 update (megamenu, hover changes, etc...), so we can close this request. Thank you!
